pkg/repository/repository.go

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2022-2023 Dell Inc, or its subsidiaries.
+
+// Package repository is the database abstraction implementing repository design pattern
+package repository
+
+// Database abstraction
+type Database interface {
```

Comment:
You might want to call it KVStore or something like that. It's not really a "database".

Reply:
You might also consider including mocked implementations of this interface to encourage that testing pattern.